### DISCUSSION TITLE
Bulk (de)select should respect `delaySelectAll`?

### DISCUSSION THREAD

**Discussion by edwinvdpol** [rappasoft/laravel-livewire-tables#2022] [General]
Hi,

I have a question regarding the bulk select checkbox.

Can you tell me why you use `$wire.clearSelected()` and `$wire.setAllSelected()` [here](https://github.com/rappasoft/laravel-livewire-tables/blob/master/resources/views/components/table/th/bulk-actions.blade.php#L22), instead of `clearSelected()` ([here](https://github.com/rappasoft/laravel-livewire-tables/blob/master/resources/js/partials/tableWrapper.js#L55)) and `setAllSelected()` ([here](https://github.com/rappasoft/laravel-livewire-tables/blob/master/resources/js/partials/tableWrapper.js#L41))?

This will prevent a re-render (depending on the `delaySelectAll` property), which I was looking for.
At this point, this property is ignored in the first place.

Also, the `delaySelectAll` should also apply to the `clearSelected()` javascript method [here](https://github.com/rappasoft/laravel-livewire-tables/blob/master/resources/js/partials/tableWrapper.js#L55), right?

Hope to hear your thoughts about this.

Best regards,
Edwin

EDIT: PS: Is [this](https://github.com/rappasoft/laravel-livewire-tables/blob/master/resources/js/laravel-livewire-tables.js#L5) javascript even used?

**Comment by lrljoe** [comment]
Now that's a good question, and one that I'll answer over the next few days, just wanted to acknowledge the question so that I don't lose track of it.

The delaySelectAll was a bit of a rush job, so yes, clearSelected does need amending to respect it, it's just not there quite yet.

But to answer - yes the laravel-livewire-tables.js code is used, is there a particular line you were trying to reference?
